python/semantic_kernel/agents/bedrock/bedrock_agent_base.py

```diff
@@ -0,0 +1,389 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import asyncio
+import logging
+from functools import partial
+from typing import Any, ClassVar
+
+import boto3
+from botocore.exceptions import ClientError
+from pydantic import Field, field_validator
+
+from semantic_kernel.agents.bedrock.action_group_utils import kernel_function_to_bedrock_function_schema
+from semantic_kernel.agents.bedrock.models.bedrock_action_group_model import BedrockActionGroupModel
+from semantic_kernel.agents.bedrock.models.bedrock_agent_model import BedrockAgentModel
+from semantic_kernel.agents.bedrock.models.bedrock_agent_status import BedrockAgentStatus
+from semantic_kernel.connectors.ai.function_choice_behavior import FunctionChoiceBehavior, FunctionChoiceType
+from semantic_kernel.kernel import Kernel
+from semantic_kernel.kernel_pydantic import KernelBaseModel
+from semantic_kernel.utils.experimental_decorator import experimental_class
+
+logger = logging.getLogger(__name__)
+
+
+@experimental_class
+class BedrockAgentBase(KernelBaseModel):
+    """Bedrock Agent Base Class to provide common functionalities for Bedrock Agents."""
+
+    # There is a default alias created by Bedrock for the working draft version of the agent.
+    # https://docs.aws.amazon.com/bedrock/latest/userguide/agents-deploy.html
+    WORKING_DRAFT_AGENT_ALIAS: ClassVar[str] = "TSTALIASID"
+
+    # Amazon Bedrock Clients
+    # Runtime Client: Use for inference
+    bedrock_runtime_client: Any
+    # Client: Use for model management
+    bedrock_client: Any
+    # Function Choice Behavior
+    # This is primarily used to control the behavior of the kernel when the agent requests functions,
+    # and to configure the kernel function action group (i.e. via filters). When this is None, users
+    # won't be able to create a kernel function action groups.
+    function_choice_behavior: FunctionChoiceBehavior = Field(default=FunctionChoiceBehavior.Auto())
+    # Agent Model
+    agent_model: BedrockAgentModel | None = None
+
+    def __init__(
```

Comment:
Forgot to include this on how instructions/prompt template config is handled, with precedence to prompt_template_config's template for instructions:

https://github.com/microsoft/semantic-kernel/blob/d0157424b4600a65ad58d699250b80c7162c17e4/python/semantic_kernel/agents/chat_completion/chat_completion_agent.py#L84C9-L99C71